packages/rrweb/src/replay/index.ts

```diff
@@ -402,9 +412,32 @@ export class Replayer {
         );
       }, 1);
     }
-    if (this.service.state.context.events.find(indicatesTouchDevice)) {
-      this.mouse.classList.add('touch-device');
-    }
+    this.service.state.context.events.forEach((e: eventWithTime) => {
+      if (indicatesTouchDevice(e)) {
+        const d = e.data as incrementalData;
+        const touchId =
+          'touchId' in d && d.touchId !== undefined
+            ? d.touchId
+            : Object.keys(this.multiTouchMap).length;
```

Comment:
I suppose it's unlikely... but what happens if we have a new touch id that comes in?

Reply:
tried to address that by adding a new function `createNewTouch` but it seems to be creating too many extra touches

Reply:
oh, that might just be because i'm on testing on replays that don't have `pointerId` actually